### DISCUSSION TITLE
Disable force-push messages on PRs or at least disable it from my view

### DISCUSSION THREAD

**Discussion by piq9117** [community/community#129935] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

The way I use git is when it's still a WIP, I commit everything into one commit with a "wip" message. Then I will keep amending or rebasing that commit as I do my work. When I'm ready, I would do a reset and do my proper commits. In my PR this would result in a wall of force-push messages like this (This is just an image i got from a google search)
![force-push-message](https://github.com/community/community/assets/12651509/49fc9e8d-4943-4e95-aa03-82970f662aa9)

This becomes especially very annoying when there's a discussion because the UI will eventually fold the discussions. Is there a way to disable force-push messages at least from my view. If other people want to see that wall of text that's on them but I want to be able to have a custom view without these messages.

**Comment by PThorpe92** [comment]
+1 Would love this feature. :+1: 

I know a few teams that have a workflow that involves a lot of force-pushing on sometimes long-standing PR's often by more than one party, and reviews/comments get buried quickly due to those.

**Comment by rdeodoo** [comment]
That would be really convenient, it really made our workflow harder than it should have been for years now.

I submitted a ticket back in 2019 (Ticket ID 268335) and a followup one in 2020 but nothing really changed since AFAIK.

<details>

<summary> 2019 request about explaining the issue with concrete public PRs example </summary>

> Do you know if something is about to be introduced about push-force logs on pull requests?
> Do you get some other remarks about that?
> The feature is really awesome, in many very specific cases, mainly to find differences quickly if a bug appeared since last force-push.
> It saved me some precious time.
> 
> But really, most of the time (I'd say 99%) you go on a pull request, that's not to see a force-push, but that's now the only thing you see..
> 
> You can have a look at https://github.com/odoo/odoo/pull/24179, I have to unfold 285 push-force notification to be able to see my review comment. Same at https://github.com/odoo/odoo/pull/33213.
> That's happening every day, a way to hide those push-force seems really mandatory, please. Something like on the PR code 'Show comments' would be more than enough.
> 
> We have some issues happening because of that, since people review a PR and did not unfold the whole push-force event and missed some feedbacks (or they did unfold but did not saw it between the hundreds of push-force).

</details>

<details>

<summary> Followup request in 2020 with other practical examples and video recording of the issue </summary>

> I wanted to know if you have anything planed to help with the push force
> notification (and maybe also commit references) on a pull request thread?
> 
> It makes the whole history and traceability of the PR completely useless
> as the useful things such as comments and/or reviews are wiped away a
> few hundreds messages over the push forces notifications.
> 
> I recorded one of the pull request I reviewed today, see
> https://drive.google.com/file/d/1pmL0tP4SProA8DsH8DUA1WUcqjwLwU6-/view
> PR: https://github.com/odoo/odoo/pull/65871
> 
> As you can see, I have to scroll to the top of the thread, unfold 6
> times the messages, to find the 3 useful review comments between
> hundreds of push forces.
> 
> I already reported it in 2019, see ticket #268335, it was sent to
> feedback but I never got an answer.
> 
> At least a button to 'hide push forces messages' or 'hide references
> related messages' that would hide push forces and or commit references.
> 
> It makes our work very harder that it should be, every push force noise the PR.

</details>

I believe they improved some stuff, because we do see less notification in the thread now, but we still see many push forces on some PRs, hidding the relevant comments.

Adding just a toggle somewhere on the UI to filter out the technical messages would be enough, as we would be able to see the comments / reviews.

**Comment by emadpres** [comment]
Also my team's workflow relies heavily on push forces.

Having an option to clear such messages before opening the PR for review (i.e., while it's in the Draft mode) would be very useful. That way, I can ensure reviewers don't get annoyed with such irrelevant messages.
